### PR TITLE
Add release types for major and minor prereleases

### PR DESCRIPTION
**Pull request by kjellknapen**
Allow user to do prereleases of major and minor releases. It's very weird that isn't a possibility right now.

### CODE REVIEW

**Comment by ecraig12345** [comment]
This makes sense but would probably be perceived as a breaking change in a lot of repos that use beachball today, since suddenly several new change types would be available and have to be manually added to their disallowedChangeTypes if not desired. I made an issue #947 to track adding this whenever we do a new major version (no specific plans for timing yet). 

In the meantime, I think a workaround would be to revert the `promptForChange` additions, but keep the new `SortedChangeTypes`. This would allow the new types to be specified as `--type` on the command line (or by editing the change file) and ensure they're handled with proper precedence. And then if you wanted to include the types in your repo's prompt, you could implement a custom `changeFilePrompt` in beachball.config.js.

**Comment by kjellknapen** [comment]
@ecraig12345 Alright, I removed the other pre* versions from the prompt. Is that something that can be added in a minor/patch release in this way or do I need to make other changes to ensure we don't break current implementation?

**Comment by kjellknapen** [comment]
I could optionally also disallow them all if prerelease is disallowed, but that may be a hacky solution?

**Comment by ecraig12345** [comment]
@kjellknapen I think it's fine if new change types are allowed by default as long as they don't show up in the prompt.

I ran a build and it looks like you'll also need to update `src/changelog/renderPackageChangelog.ts` `groupNames` with the new types. Those are used as headers in the CHANGELOG.md if any changes of that type are present.

**Comment by kjellknapen** [comment]
@ecraig12345 Done! Thanks for the quick response. I used the same title as the respective versions but added (pre-release)  behind them.

**Comment by ecraig12345** [comment]
Looks like you accidentally installed with pnpm (this repo uses yarn)--can you delete the pnpm lock file?

**Comment by kjellknapen** [comment]
Oops my bad, was working in a different project with pnpm, so I was in that flow. I removed it

**Review by ecraig12345 — DISMISSED** [review]

**Comment by ecraig12345** [comment]
Oh I guess you need a change file too.

**Review by ecraig12345 — DISMISSED** [review]

**Comment by kjellknapen** [comment]
I guess I need to update some tests aswell

File: src/__tests__/changefile/changeTypes.test.ts
Comment (kjellknapen) [inline src/__tests__/changefile/changeTypes.test.ts:1]: @ecraig12345  Does the test failing here signal that my changes are still breaking?

**Review by ecraig12345 — APPROVED** [review]

**Pull request merged**